sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -329,13 +329,13 @@ namespace Azure { namespace Core { namespace Http {
     const std::string& GetPath() const { return m_encodedPath; }
 
     /**
-     * @brief Provides the list of query parameters from the URL.
+     * @brief Provides a copy to the list of query parameters from the URL.
      *
      * @remark The query parameters are URL-encoded.
      *
      * @return const std::map<std::string, std::string>&
      */
-    const std::map<std::string, std::string>& GetQueryParameters() const
+    const std::map<std::string, std::string> GetQueryParameters() const
```

Comment:
why do you return a copy?

Reply:
what's the benefit of returning a copy over returning a ref?

Reply:
got feedback from the team (@ahsonkhan , @RickWinter )
We don't want to expose the internal structure out of URL because if the URL is reassigned or destroyed, then the reference becomes a possible bug and corrupted memory.
A copy will keep valid even if the URL is gone.

I also prefer the reference, for perf reasons,, and because usually anyone getting a ref to an object knows about the lifetime of that ref depends on something else that need to be keep while using the ref.  But it is still non-error probe

Reply:
> We don't want to expose the internal structure out of URL because if the URL is reassigned or destroyed, then the reference becomes a possible bug and corrupted memory.

This doesn't make sense at all. Then how do you explain [this](https://github.com/Azure/azure-sdk-for-cpp/blob/7c3257879870369560358eb35b664f7a75b5d269/sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp#L75) API?

Reply:
I'm okay with either by value or by const reference. But why const value? Why const is necessary?

Reply:
you are right, for the copy, we don't need the const.
thank you